components/script/serviceworkerjob.rs

```diff
@@ -183,69 +187,36 @@ impl JobQueue {
             job_vec.unwrap().first().unwrap()
         };
         let global = &*front_job.client.global();
+        let handler = *run_job_handler;
         match front_job.job_type {
             JobType::Register => {
-                let handler = *run_job_handler;
                 let register_job_handler = AsyncJobHandler::new(handler.scope_url, InvokeType::Register);
-                script_thread.queue_register_job(box register_job_handler, global);
+                script_thread.queue_serviceworker_job(box register_job_handler, global);
             },
-            _ => {}
-        }
-    }
-
-    #[allow(unrooted_must_root)]
-    pub fn settle_promise_job(&self, async_job_handler: Box<AsyncJobHandler>, script_thread: &ScriptThread) {
-        use self::InvokeType::*;
-        let queue_ref = &*self.0.borrow();
-        let front_job =  {
-            let job_vec = queue_ref.get(&async_job_handler.scope_url);
-            job_vec.unwrap().first().unwrap()
-        };
-        let handler = *async_job_handler;
-        let scope_url = front_job.scope_url.clone();
-        let invoke_type = handler.invoke_type;
-        let global = &*front_job.client.global();
-        let trusted_global = Trusted::new(global);
-        let promise = &front_job.promise;
-        let _ac = JSAutoCompartment::new(global.get_cx(), promise.reflector().get_jsobject().get());
-        match invoke_type {
-            Settle(SettleType::Resolve(reg)) => {
-                let reg = reg.root();
-                promise.resolve_native(global.get_cx(), &*reg);
-                let finish_job_handler = box FinishJobHandler {
-                    scope_url: scope_url,
-                    global: trusted_global
-                };
-                script_thread.queue_finish_job(finish_job_handler, global);
-            }
-            Settle(SettleType::Reject(err_type)) => {
-                promise.reject_error(global.get_cx(), err_type);
-                let finish_job_handler = box FinishJobHandler {
-                    scope_url: scope_url,
-                    global: trusted_global
-                };
-                script_thread.queue_finish_job(finish_job_handler, global);
+            JobType::Update => {
+                let update_job_handler = AsyncJobHandler::new(handler.scope_url, InvokeType::Update);
+                script_thread.queue_serviceworker_job(box update_job_handler, global);
             }
-            _ => warn!("other variants already matched before")
+            _ => { unimplemented!(); }
```

Comment:
No, as of now, we don't have any code that creates a Job of type `Unregister`. Anyways, should we replace it with a todo comment instead ?